fairmq/plugins/DDS/runDDSCommandUI.cxx

```diff
@@ -49,279 +49,155 @@ struct TerminalConfig
     }
 };
 
-struct StateSubscription
-{
-    const string& fTopologyPath;
-    CCustomCmd& fDdsCustomCmd;
-
-    explicit StateSubscription(const string& topologyPath, CCustomCmd& ddsCustomCmd)
-        : fTopologyPath(topologyPath)
-        , fDdsCustomCmd(ddsCustomCmd)
-    {
-        fDdsCustomCmd.send(Cmds(make<SubscribeToStateChange>()).Serialize(), fTopologyPath);
-    }
-
-    ~StateSubscription() {
-        fDdsCustomCmd.send(Cmds(make<UnsubscribeFromStateChange>()).Serialize(), fTopologyPath);
-        this_thread::sleep_for(chrono::milliseconds(100)); // give dds a chance to complete request
-    }
-};
-
 void printControlsHelp()
 {
     cout << "Use keys to control the devices:" << endl;
     cout << "[c] check states, [o] dump config, [h] help, [r] run, [s] stop, [t] reset task, [d] reset device, [q] end, [j] init task, [i] init device, [k] complete init, [b] bind, [x] connect" << endl;
     cout << "To quit press Ctrl+C" << endl;
 }
 
-void sendCommand(const string& commandIn, const string& topologyPath, CCustomCmd& ddsCustomCmd)
+void handleCommand(const string& command, const string& path, unsigned int timeout, Topology& topo)
 {
-    char c;
-    string command(commandIn);
-    TerminalConfig tconfig;
-
-    if (command == "") {
+    if (command == "c") {
+        cout << "> checking state of the devices" << endl;
+        topo.GetCurrentState();
+        // TODO: extend me
+    } else if (command == "o") {
+        cout << "> dumping config of the devices" << endl;
+        auto const result = topo.GetProperties("^(session|id)$", path, std::chrono::milliseconds(timeout));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'result' is assigned a value that is never used.](https://app.codacy.com/manual/FairRootGroup/FairMQ/pullRequest?prid=5000642)